rcl/src/rcl/time.c

```diff
@@ -12,8 +12,308 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <stdlib.h>
+
 #if defined(WIN32)
 #include "./time_win32.c"
 #else  // defined(WIN32)
 #include "./time_unix.c"
 #endif  // defined(WIN32)
+
+#include "./stdatomic_helper.h"
+
+// Process default ROS time sources
+static rcl_time_source_t * rcl_default_ros_time_source;
+static rcl_time_source_t * rcl_default_steady_time_source;
+static rcl_time_source_t * rcl_default_system_time_source;
+
+
+/** Internal storage for RCL_ROS_TIME implementation */
+typedef struct
+{
+  atomic_uint_least64_t current_time;
+  bool active;
+  // TODO(tfoote): store subscription here
+} rcl_ros_time_source_storage_t;
+
+/** Implementation only **/
+rcl_ret_t rcl_get_steady_time(void * data, rcl_time_point_imp_t * current_time)
+{
+  (void) data; /* unused */
```

Comment:
There are still uses of this in:
* rcl/src/rcl/time.c
* rcl/test/rcl/test_time.cpp